lib/Table/TableAutomaticStylesStratum.ts

```diff
@@ -258,11 +260,24 @@ export class ColorStyleLegend extends LoadableStratum(LegendTraits) {
           ]
         : [];
 
-    return colorMap.values
-      .map((value, i) => {
+    // Aggregate colours (don't show multiple legend items for the same colour)
+    const colorMapValues = colorMap.values.reduce<{
+      [color: string]: string[];
+    }>((prev, current, i) => {
+      const cssCol = colorMap.colors[i].toCssColorString();
+      if (isDefined(prev[cssCol])) {
+        prev[cssCol].push(current);
+      } else {
+        prev[cssCol] = [current];
+      }
+      return prev;
+    }, {});
+
+    return Object.entries(colorMapValues)
+      .map(value => {
         return createStratumInstance(LegendItemTraits, {
-          title: value,
-          color: colorMap.colors[i].toCssColorString()
+          multipleTitles: value[1],
```

Comment:
Could you put a comment in here explaining why you're using these particular `indices` to the value array?

Reply:
It is from `Object.entries(colorMapValues)`, so `value` is `[key, value]`

Reply:
It is from `Object.entries(colorMapValues)`, so `value` is `[key, value]`

Reply:
OH duh sorry of course